presentation/backend/src/presentation-backend/PresentationManagerDetail.ts

```diff
@@ -153,7 +155,12 @@ export class PresentationManagerDetail implements IDisposable {
     }
 
     const response = await this.getNativePlatform().handleRequest(imodelAddon, JSON.stringify(nativeRequestParams), cancelEvent);
-    diagnosticsListener && response.diagnostics && diagnosticsListener({ logs: [response.diagnostics] });
+    if (response.diagnostics) {
+      const logs = { logs: [response.diagnostics] };
+      diagnosticsListener && diagnosticsListener(logs);
+      this._diagnosticsCallback && this._diagnosticsCallback(convertToReadableSpans(logs));
```

Comment:
I think  `this._diagnosticsCallback` should accept `Diagnostics` rather than `ReadableSpan[]` as its argument.